message_definitions/v1.0/ras_a.xml

```diff
@@ -10,7 +10,140 @@
+           If the passed exploration ID does not exist or is not listed as a valid exploration ID in the vehicle autonomy engine, then this command should be rejected.
+        </description>
+        <param index="1" label="Exploration task ID" minValue="0" maxValue="4294967295" increment="1">Sets the ID of the (current or queued) exploration task to stop. Set to UINT32_MAX if one wants to stop the current exploration task.</param>
+        <param index="2" label="Behavior after stopping" minValue="0" maxValue="2" increment="1">The behavior after stopping the exploration task. 0: Hold, 1: Land, 2: Return to position.</param>
+        <param index="3" reserved="true" default="NaN"/>
+        <param index="4" reserved="true" default="NaN"/>
+        <param index="5" reserved="true" default="NaN"/>
+        <param index="6" reserved="true" default="NaN"/>
+        <param index="7" reserved="true" default="NaN"/>
+      </entry>
+      <entry value="40512" name="MAV_CMD_SET_EXPLORATION_RETURN_POS" hasLocation="true" isDestination="false">
+        <wip/>
+        <!-- This command is work-in-progress and it can therefore change. It should NOT be used in stable production environments. -->
+        <description>
+          Sets the return position after stopping or finishing an exploration task. Used when the vehicle autonomy engine does not set this position or to overwrite it.
+          COMMAND_INT should be used so to set the MAV_FRAME and consequently, frame and coordinates of the position (MAV_FRAME_GLOBAL means global coordinates are being used,
+          while MAV_FRAME_LOCAL_NED means local NED coordinates are being used).
+        </description>
+        <param index="1" reserved="true" default="0"/>
+        <param index="2" reserved="true" default="0"/>
+        <param index="3" reserved="true" default="0"/>
+        <param index="4" label="Yaw or Heading" units="rad">Yaw or heading of the return position. NaN if unknown.</param>
+        <param index="5" label="Position X or Latitude">X (mE4) or Latitude (degE7) (WGS84) coordinate of the return position. NaN or INT32_MAX if unknown.</param>
+        <param index="6" label="Position Y or Longitude">Y (mE4) or Longitude (degE7) (WGS84) coordinate of the return position. NaN or INT32_MAX if unknown.</param>
+        <param index="7" label="Position Z or Altitude (MSL)" units="m">Z or Altitude (MSL) coordinate of the return position. NaN or INT32_MAX if unknown.</param>
+      </entry>
+      <entry value="40513" name="MAV_CMD_DO_EXPLORATION_RETURN" hasLocation="false" isDestination="false">
+        <wip/>
+        <!-- This command is work-in-progress and it can therefore change. It should NOT be used in stable production environments. -->
+        <description>Return to a system-defined position after exploration.</description>
```

Comment:
Just FMI if sent during when in the middle of the building, is the assumption the vehicle will follow some path or just head straight through the walls, if needed?

Reply:
Nova 2 follows a safe path back to the return point. This command is pretty useless otherwise, so perhaps that should be specified.

Reply:
I think so. If the safe path back might be controllable in future you might even be specific and/or make this an option. Otherwise you end up having to design it in later, in the say way PX4 support RTL, rally points, mission landing, mission landing reverse (and several more).

Is it a valid assumption that there will always have a safe path back that the vehicle is able to follow? Can you lose positioning data? I'm not saying there isn't  - mostly just interested.

Reply:
Of course, you can never assume that anything is safe in all situations. The question is the intent of the command. It should be specified that the intent is that exploration return is safe from indoors, otherwise the command has no purpose.

Reply:
Cool. I'm not certain whether we need to say safe path or not - simply because I don't think anyone would assume that the drone would attempt to go through walls. It might be worth saying ("following a system-defined path") - to make it clear that this is not something you can or need to specify.